vm.go

```
@@ -855,6 +855,23 @@ func (vm *vm) push(v Value) {
 	vm.stack.expand(vm.sp)
 	vm.stack[vm.sp] = v
 	vm.sp++
+
+	shouldTrackMaxMemOnStack := (vm.r != nil && vm.r.shouldTrackMaxMemOnStack)
+
+	if !shouldTrackMaxMemOnStack || v == nil || !v.IsObject() {
+		return
+	}
+
+	baseObjectClass := v.baseObject(vm.r).Class()
+
+	// Function and Arguments objects appear to contain references to other objects on the stack.
+	// Checking just these two types will be an efficient, but still accurate, memory estimate over checking all objects
+	if baseObjectClass != "Function" && baseObjectClass != "Arguments" {
+		return
+	}
+
+	valueMemUsage, _ := v.MemUsage(vm.r.stackMemUsageContext)
```

[minor] Although it rarely happens, what should we expect if v.MemUsage returns an error?

---

Ah yeah, I'll add a comment saying that we will swallow the error. I didn't want to thread the logger through here for now since I don't see anywhere that err would ever be set to non-nil. Though I could be missing it